addons/website/static/tests/builder/website_builder/menu_data.test.js

```diff
@@ -195,4 +197,103 @@ describe("EditMenuDialog", () => {
         expect(".oe_menu_editor").toHaveCount(1);
         expect(".js_menu_label").toHaveText("Top Menu Item");
     });
+
+    test("clicking save in the EditMenuDialog should not clear the editor changes", async () => {
+        const { getEditor } = await setupWebsiteBuilder(
+            // Using tel: as link to avoid having to mock fetching metadata for link preview
+            // This does not influence the test in any way
+            `<ul class="top_menu">
+                <li>
+                    <a class="nav-link" href="tel: 123" contenteditable="true">
+                        <span>Top Menu Item</span>
+                    </a>
+                </li>
+            </ul>
+            <section>
+                <p>Text</p>
+            </section>`,
+            {
+                openEditor: true,
+            }
+        );
+
+        const fakeORMService = {
+            call: (model, method, args) => {
+                if (method === "save" && model === "ir.ui.view") {
+                    expect(args[1]).toInclude("Text, more text");
+                    hasSaved = true;
+                }
+
+                return {
+                    fields: {
+                        id: 4,
+                        name: "Top Menu",
+                        url: "#",
+                        new_window: false,
+                        is_mega_menu: false,
+                        sequence: 0,
+                        parent_id: false,
+                    },
+                    children: [
+                        {
+                            fields: {
+                                id: 5,
+                                name: "Top Menu Item",
+                                url: "tel: 123",
+                                new_window: false,
+                                is_mega_menu: false,
+                                sequence: 10,
+                                parent_id: 4,
+                            },
+                            children: [],
+                            is_homepage: true,
+                        },
+                    ],
+                    is_homepage: false,
+                };
+            },
+        };
```

Comment:
Why not use onRPC ? I think it s better way to mock something.
1 onRPC for the save
1 onRPC for the other scenario